### PR TITLE
Check for `clientId` before presence publishing attempt

### LINKED ISSUE #1126: SDK client should automatically use clientId from the tokenParams or TokenDetails (RSA7b2)
I don't think ably-cocoa supports [RSA7b2](https://docs.ably.io/client-lib-development-guide/features/#RSA7b2)

> (RSA7b) Auth#clientId is not null when:
  > (RSA7b2) Token authentication is being used, and the TokenRequest or TokenDetails object, used for authentication, has a clientId value that is not null

If I were to summarize: SDK client should automatically use clientId from the tokenParams or TokenDetails if `clientId` is set to null in clientOptions. I believe this is clearer.

┆Issue is synchronized with this [Jira Story](https://ably.atlassian.net/browse/SDK-919) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by maratal**
Closes #1126

### CODE REVIEW

**Comment by lawrence-forooghian** [comment]
Hey @maratal, I'm a bit confused about what this PR is doing. It says that it closes #1126, but looking at #1126 it's not clear to me what is the actual work that needs to be done there, if any. And this PR doesn't introduce any new tests so I guess it's not addressing any additional spec points.

Based on https://github.com/ably/ably-asset-tracking-swift/issues/401#issuecomment-1286010092, I had thought that perhaps you might be opening a PR for addressing spec point RTP8j, as added in https://github.com/ably/specification/pull/106. Maybe that's what this PR does, but it doesn't state that, and it doesn't add any tests for it.

**Comment by maratal** [comment]
> Based on https://github.com/ably/ably-asset-tracking-swift/issues/401#issuecomment-1286010092, I had thought that perhaps you might be opening a PR for addressing spec point RTP8j, as added in https://github.com/ably/specification/pull/106. Maybe that's what this PR does, but it doesn't state that, and it doesn't add any tests for it.

Hey @lawrence-forooghian, sorry for not being clear about this, but what this PR is doing was "inspired" (copied) by ably-java implementation, since you've mentioned it in [this comment](https://github.com/ably/ably-asset-tracking-swift/issues/401#issuecomment-1286010092). I believe that these changes reflect new spec point RTP8j, because this new spec point in its turn (as I understood), was also taken from the ably-java behavior. Re tests, you are right, I was a little too hasty to invite reviewers, because I was excited that these changes actually fixed an original issue in AAT. That's why I've converted this PR back to draft. Sorry for early disturbance, but since you are already here I appreciate your thoughts about these changes.

File: Source/ARTRealtimePresence.m [hunk 1]
```diff
@@ -500,14 +500,16 @@ - (void)addPendingPresence:(ARTProtocolMessage *)msg callback:(ARTStatusCallback
...
-    if (!_channel.realtime.connection.isActive_nosync) {
-        if (callback) callback([_channel.realtime.connection error_nosync]);
-        return;
+    NSString *authClientId = _channel.realtime.auth.clientId_nosync;
+    BOOL connected = _channel.realtime.connection.state_nosync == ARTRealtimeConnected;
+    
+    if (connected && (authClientId == nil || [authClientId isEqualToString:@"*"])) {
+        if (msg.clientId == nil) {
+            if (callback) {
+                callback([ARTErrorInfo createWithCode:ARTStateNoClientId message:@"Invalid attempt to enter without clientId."]);
```
[the remark this answers is not in the document]
Comment (lawrence-forooghian) [reply Source/ARTRealtimePresence.m:510]: Implemented in #1515.

File: Source/ARTRealtimePresence.m [hunk 2]
```diff
@@ -500,14 +500,16 @@ - (void)addPendingPresence:(ARTProtocolMessage *)msg callback:(ARTStatusCallback
...
-        return;
-    }
-
-    if (!_channel.realtime.connection.isActive_nosync) {
-        if (callback) callback([_channel.realtime.connection error_nosync]);
-        return;
+    NSString *authClientId = _channel.realtime.auth.clientId_nosync;
+    BOOL connected = _channel.realtime.connection.state_nosync == ARTRealtimeConnected;
+    
+    if (connected && (authClientId == nil || [authClientId isEqualToString:@"*"])) {
```
[the remark this answers is not in the document]
Comment (lawrence-forooghian) [reply Source/ARTRealtimePresence.m:507]: Implemented in #1515.

**Comment by lawrence-forooghian** [comment]
@maratal other than what's covered in #1517, is there anything else that this PR addresses that you think needs a separate PR / issue, or can we close this one?

**Comment by maratal** [comment]
> @maratal other than what's covered in #1517, is there anything else that this PR addresses that you think needs a separate PR / issue, or can we close this one?

If tests pass in AAT with changes only in #1517, then this one should be closed.

**Pull request closed without merging**